ntbs-service/DataAccess/NotificationRepository.cs

```diff
@@ -71,6 +72,11 @@ public bool NotificationWithLegacyIdExists(string id)
                 .Any(e => e.LTBRID == id || e.ETSID == id);
         }
 
+        public async Task<int> GetNumberOfNotificationsInCluster(string clusterId)
+        {
+            return await Task.Run(() => _context.Notification.Count(n => n.ClusterId == clusterId));
```

Comment:
I would suggest using `await _context.Notification.CountAsync()` currently this is artificially async.